compiler/src/iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp

```diff
@@ -440,6 +440,42 @@ class TransposeReshapeGenericDotGeneral
   }
 };
 
+struct ScatterInt64Indices : public OpRewritePattern<mhlo::ScatterOp> {
+  using OpRewritePattern<mhlo::ScatterOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(mhlo::ScatterOp op,
+                                PatternRewriter &rewriter) const final {
+    auto indices = op.getScatterIndices();
+    auto indicesTy = indices.getType();
+    auto indicesETy = indicesTy.getElementType();
+    if (indicesETy.isInteger(32))
+      return rewriter.notifyMatchFailure(op, "already has i32 index type");
+
+    if (!indicesTy.hasStaticShape())
+      return rewriter.notifyMatchFailure(op, "cannot validate legal size");
+
+    uint64_t maxSize = std::numeric_limits<int32_t>::max();
+    if (indicesETy.getIntOrFloatBitWidth() > 32)
+      for (int i = 0, s = indicesTy.getRank(); i < s; ++i)
+        if (indicesTy.getDimSize(i) > maxSize)
+          return rewriter.notifyMatchFailure(op, "index may exceed i32 max");
```

Comment:
IREE style nit: wrap with braces on if/for simple single-statement bodies.

Reply:
Done.